tribe-ext-limit-week-view-time-range.php

```diff
@@ -0,0 +1,315 @@
+ *
+ *     This plugin is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+namespace Tribe\Extensions\Limit_Week_View_Time_Range;
+
+use Tribe__Autoloader;
+use Tribe__Extension;
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if ( class_exists( 'Tribe__Extension' ) && ! class_exists( Main::class ) ) {
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Main extends Tribe__Extension {
+
+		/**
+		 * @var Tribe__Autoloader
+		 */
+		private $class_loader;
+
+		/**
+		 * @var Settings
+		 */
+		private $settings;
+
+		/**
+		 * Setup the Extension's properties.
+		 *
+		 * This always executes even if the required plugins are not present.
+		 */
+		public function construct() {
+			$this->add_required_plugin( 'Tribe__Events__Pro__Main', '5.0' );
+		}
+
+		/**
+		 * Get this plugin's options prefix.
+		 *
+		 * Settings_Helper will append a trailing underscore before each option.
+		 *
+		 * @return string
+		 * @see \Tribe\Extensions\Limit_Week_View_Time_Range\Settings::set_options_prefix()
+		 *
+		 */
+		private function get_options_prefix() {
+			return (string) str_replace( '-', '_', 'tribe-ext-limit-week-view-time-range' );
+		}
+
+		/**
+		 * Get Settings instance.
+		 *
+		 * @return Settings
+		 */
+		private function get_settings() {
+			if ( empty( $this->settings ) ) {
+				$this->settings = new Settings( $this->get_options_prefix() );
+			}
+
+			return $this->settings;
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			// Load plugin textdomain
+			load_plugin_textdomain( 'tribe-ext-limit-week-view-time-range',
+			                        false,
+			                        basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			if ( ! $this->php_version_check() ) {
+				return;
+			}
+
+			/*if ( ! $this->is_using_compatible_view_version() ) {
+				return;
+			}*/
+
+			$this->class_loader();
+
+			$this->get_settings();
+
+			add_filter( 'tribe_events_week_get_hours', [ $this, 'filter_week_hours' ] );
+			add_filter( 'tribe_template_path_list', [ $this, 'alternative_week_view_template_locations' ], 10, 2 );
+			add_action( 'wp_enqueue_scripts', [ $this, 'enquque_styles' ] );
+		}
+
+		/**
+		 * Template override in plugin folder. Needed for V2.
+		 *
+		 * @param $folders
+		 * @param \Tribe__Template $template
+		 *
+		 * @return mixed
+		 */
+		function alternative_week_view_template_locations( $folders, \Tribe__Template $template ) {
+			// Which file namespace your plugin will use.
+			$plugin_name = 'tribe-ext-limit-week-view-time-range';
+
+			// Which order we should load your plugin files at.
+			$priority = 5;
+			// Plugin in which the file was loaded from = 20
+			// Events Pro = 25
+			// Tickets = 17
```

Comment:
What does "Plugin in which the file was loaded from" mean?

```suggestion
			// Which order we should load your plugin files at. Plugin in which the file was loaded from = 20. Events Pro = 25. Tickets = 17.
			$priority = 5;
```

Reply:
TBH I don't know. It comes from the example from our kb.
https://theeventscalendar.com/knowledgebase/k/custom-templates-for-the-updated-calendar-views/